herd/AArch64Sem.ml

```diff
@@ -1407,7 +1422,25 @@ module Make
             M.unitT B.Next
               (* Branches *)
         | I_B l ->
-            B.branchT l
+            if not self then
+              B.branchT l
+            else begin
+              match Label.Set.elements ii.A.labels with
+              | hd::_ ->
+                  let instr_v = A.V.cstToV (A.instruction_to_value ii.A.inst) in
+                  M.altT  (
+                    read_loc_instr hd ii
+                    >>= is_eq instr_v
+                    >>= is_not_zero
+                    >>= fun _ -> B.branchT l
```

Comment:
Here the instruction is fetched compared with instr_v and the result is thrown away (line `fun _ -> ...). Instead one should add a constraint that the fetched instruction equals instr_v:
```suggestion
                    >>= M.eqT instr_v
                    >>= fun () -> B.branchT l
```

Reply:
Applied, thanks.